org.mwc.debrief.core/src/org/mwc/debrief/core/creators/shapes/CoreInsertShape.java

```diff
@@ -118,12 +123,35 @@ protected String getLayerName()
             if (dlg.open() == Window.OK)
             {
               res = dlg.getValue();
-              // create base layer
-              final Layer newLayer = new BaseLayer();
-              newLayer.setName(res);
-
-              // add to layers object
-              theLayers.addThisLayer(newLayer);
+           
+              final String title = "Forbidden Name for Layer";
+              final String messageNarratives = "\'Narratives\' is a reserved Layer name";
+              final String messageNarratives2 = "Choose a different name for your layer";
+              final String messageCannotBeEmpty = "Message cannot be empty";
+              final String messageCannotBeEmpty2 = "You need to have at least one character in your layer name";
+              if ( res == null || res.isEmpty() )
+              {
+                Status status = new Status(IStatus.ERROR, Activator.PLUGIN_ID,  messageCannotBeEmpty2, new Exception(""));
+
+                ErrorDialog.openError(Display.getCurrent().getActiveShell(), title, messageCannotBeEmpty, status);
+                
+                res = null;
+              }else if ( "narratives".equals(res.toLowerCase()) )

```

Comment:
let's not use a magic string, let's use the constant that's defined in `NarrativeEntry`

Reply:
oh, ok ok